packages/react-native-gesture-handler/src/v3/types/EventTypes.ts

```diff
@@ -49,12 +49,9 @@ export type TouchEvent =
   | GestureTouchEvent
   | NativeSyntheticEvent<GestureTouchEvent>;
 
-export type GestureStateChangeEvent<THandlerData> = StateChangeEventPayload &
-  THandlerData;
-export type GestureUpdateEvent<THandlerData> = EventPayload & THandlerData;
+export type GestureUpdateEvent<THandlerData> = THandlerData;
```

Comment:
I'm not sure if this won't result in some name conflicts, but what about making it `GestureEvent` instead?

Reply:
Done in https://github.com/software-mansion/react-native-gesture-handler/pull/3844/commits/f30f2f010a2e1398bb147138e4376b97d7ae3821